wikimedia/wikipedia-ios
# Pass edit preview parameter in non-local environments

**Fixes Phabricator ticket:** N/A

### Notes
Well, this is why one should keep local and production code identical. 
- To allow testing of edit preview while scheme is set to local PCS, we had to fork the logic.
- We then went back and added code to send a parameter for the edit preview type...
- ...but only did that for the code for local PCS, not the production code.
- (This is yet to rear it's ugly head as a live bug because the server-side code is yet to deploy. It will be held for this, so no time crunch on getting this in.)

While working on this, I tried to write code to simplify testing this locally - some extra code for if you're running local restbase as well as local PCS. This ended up with more forking code running within the forking code, but this new code having to be setup in the `ArticleFetcher`. And assuming that your local restbase port is always 7231. Ultimately, given how edge case it was, adding more code complexity to help testing a corner case seemed like overengineering, (especially because the existing code complexity around this is precisely what led to this bug). I didn't include that code in the PR.

### Test Steps
Therein lies the rub. This is hard to test. You can setup parsoid and restbase locally, add some custom code on restbase (otherwise the second call in the wikitext-to-mobilehtml cocktail won't work), and test it... But that's a lot of work for not actually testing the proper route.

My suggestion, feel free to push back:
1. Test edit preview* on local PCS scheme.
2. Test edit preview* on production scheme.
3. After [this restbase patch](https://gerrit.wikimedia.org/r/c/mediawiki/services/restbase/deploy/+/605650) gets deployed, I'll test both those again.

*Edit an article, see the preview of your edit, ensure edit pencil icons are not shown.
